src/peerstate.rs

```diff
@@ -457,59 +457,44 @@ impl Peerstate {
         Ok(backward_verified)
     }
 
-    /// Set this peerstate to verified
-    /// Make sure to call `self.save_to_db` to save these changes
+    /// Set this peerstate to verified;
+    /// make sure to call `self.save_to_db` to save these changes.
+    ///
     /// Params:
-    /// verifier:
+    ///
+    /// * key: The new verified key.
+    /// * fingerprint: Only set do verified if the key's fingerprint matches this.
+    /// * verifier:
     ///   The address which introduces the given contact.
     ///   If we are verifying the contact, use that contacts address.
     pub fn set_verified(
         &mut self,
-        which_key: PeerstateKeyType,
+        key: SignedPublicKey,
         fingerprint: Fingerprint,
```

Comment:
I'm not completely sure whether we still need the `fingerprint` parameter. We could also remove it and unconditionally set the `verified_key` - I can't come up with a scenario where this would be a problem, but also I'm not sure I thought of every possible scenario.

Reply:
It's still needed -- see the `verify_sender_by_fingerprint()` call for Alice in the `Steps 5+6 in "Setup verified contact" protocol` section, in that case the fingerprint must be taken from the Secure-Join-Fingerprint header

Reply:
Since `verify_sender_by_fingerprint()` first loads the key _with this fingerprint_ and only then calls `set_verified()`, I don't think that this particular usage of `set_verified()` requires the `fingerprint` parameter?